$MirOS: ports/net/djbdns/patches/patch-src_dnsfilter_c,v 1.1.201.1 2005/05/14 23:21:03 tg Exp $
--- src/dnsfilter.c.orig	Sun Feb 11 21:11:23 2001
+++ src/dnsfilter.c	Tue Dec 13 21:44:24 2005
@@ -12,6 +12,9 @@
 #include "iopause.h"
 #include "error.h"
 #include "exit.h"
+#ifdef DJB_V6ONLY
+#include "ip6.h"
+#endif /* DJB_V6ONLY */
 
 #define FATAL "dnsfilter: fatal: "
 
@@ -44,7 +47,11 @@ int flag0 = 1;
 iopause_fd *io;
 int iolen;
 
+#ifndef DJB_V6ONLY
 char servers[64];
+#else
+char servers[256];
+#endif /* DJB_V6ONLY */
 char ip[4];
 char name[DNS_NAME4_DOMAIN];
 
@@ -191,7 +198,11 @@ int main(int argc,char **argv)
 	      dns_name4_domain(name,ip);
 	      if (dns_resolvconfip(servers) == -1)
 	        strerr_die2sys(111,FATAL,"unable to read /etc/resolv.conf: ");
+#ifndef DJB_V6ONLY
 	      if (dns_transmit_start(&x[xnum].dt,servers,1,name,DNS_T_PTR,"\0\0\0\0") == -1)
+#else
+	      if (dns_transmit_start(&x[xnum].dt,servers,1,name,DNS_T_PTR,V6any) == -1)
+#endif /* DJB_V6ONLY */
 	        errout(xnum);
 	      else {
 	        x[xnum].flagactive = 1;
